### PR TITLE
Add metrics

### PR DESCRIPTION
**Pull request by joecohens**
To finish metrics we need:

- [x] Add an interval for the graph
- [x] Add metric points through the API
- [x] Display metric points on graph

### CODE REVIEW

**Comment by jbrooksuk** [comment]
Rebased to fix issues with Composer.

**Comment by jbrooksuk** [comment]
@GrahamCampbell don't kill me, I've been messing around just getting it to work for now.

With my most recent changes we get:

image

**Comment by jbrooksuk** [comment]
Over to you @joecohens to check I've not missed anything and then we should merge.

File: src/Http/Controllers/HomeController.php [hunk 1]
```diff
@@ -66,6 +67,10 @@ public function showIndex()
             }
         }
 
+        if (Setting::get('display_graphs')) {
+            $metrics = Metric::where('display_chart', 1)->get();
+        }
```
Comment (joecohens) [inline src/Http/Controllers/HomeController.php]: If you don't have enabled metrics will throw `Undefined variable: metrics`

File: bower.json [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
-        "messenger": "~1.4.1",
-        "Sortable": "~1.0.0",
-        "animate-sass": "~0.6.2",
-        "moment": "~2.9",
-        "livestampjs": "~1.1.2"
-    },
-    "resolutions": {
-        "jquery": "~2.1.1"
-    }
+  "name": "Cachet",
```
Comment (joecohens) [inline bower.json]: Indent this to 4 spaces?

File: src/Http/Controllers/HomeController.php (hunk 1 above)
Comment (jbrooksuk) [inline src/Http/Controllers/HomeController.php]: Ah crap.

File: src/Http/Controllers/DashMetricController.php [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+                ))
+                ->with('errors', $point->getErrors());
+        }
+
+        $successMsg = sprintf(
+            '<strong>%s</strong> %s',
+            trans('dashboard.notifications.awesome'),
+            trans('dashboard.metrics.points.add.success')
+        );
+
```
Comment (joecohens) [inline src/Http/Controllers/DashMetricController.php]: We need to add the segment tracking on here too

Comment (jbrooksuk) [inline src/Http/Controllers/DashMetricController.php]: Good point!

**Comment by jbrooksuk** [comment]
All done. Merge away :)

**Pull request merged**